colab-chat/colab-server
# Use uwsgi server

This makes use of uwsgi and nginx to serve colab. Based on this we can make use of async frameworks which are required for `Flask-sse` have #6. This PR also makes use of Docker to manage upcoming dependencies such as Redis etc.

Nginx should allow us to serve to several colab_server instances on a single machine. 

Useful links are:
http://www.patricksoftwareblog.com/how-to-configure-nginx-for-a-flask-web-application/
https://github.com/miguelgrinberg/microflack_messages

Look at:  https://github.com/jazzdd86/alpine-flask for enabling debug mode

## To Test

1. Create a database with `python ./web/manage.py create_db
1. run `docker-compose up`
1. Confirm that colab is being served at `localhost:80`
